src/Serilog.Sinks.MSSqlServer/Sinks/MSSqlServer/MSSqlServerSink.cs

```diff
@@ -104,10 +109,27 @@ public class MSSqlServerSink : PeriodicBatchingSink
                 using (var copy = new SqlBulkCopy(cn))
                 {
                     copy.DestinationTableName = _tableName;
-                    await copy.WriteToServerAsync(_eventsTable, _token.Token);
-
-                    // Processed the items, clear for the next run
-                    _eventsTable.Clear();
+                    try
+                    {
+                        await copy.WriteToServerAsync(_eventsTable, _token.Token);
+                    }
+                    catch (Exception ex)
+                    {
+                        // create LogEvent so exception can be added to DB during next batch run.
+                        var logEvent = new LogEvent(
+                            DateTimeOffset.Now,
+                            LogEventLevel.Error,
+                            ex,
+                            new MessageTemplate("", new[] {new TextToken("Internal error while trying to write events to database")}),
+                            Enumerable.Empty<LogEventProperty>());
+
+                        Emit(logEvent);
```

Comment:
I do agree with some additional error handling and clearing the events table. But submitting another event can blow this thing up at some point.